flax/training/checkpoints.py

```diff
@@ -313,3 +315,62 @@ def __call__(self, x):
     params_renamed[name] = convert_pre_linen(value)
 
   return core.freeze(params_renamed)
+
+
+class TrainState(struct.PyTreeNode):
```

Comment:
I propose putting this in a new file as it can be used independently of checkpointing.

Reply:
Moved to `flax.training.training_state`. Don't hesitate to inspire better naming if you can...